CI/functional_tests/test_molten_salts.py

```diff
@@ -90,18 +95,59 @@ def mdsuite_project(traj_files, tmp_path) -> mds.Project:
     return project
 
 
-def test_analysis():
+def test_analysis(mdsuite_project):
     """
     Perform analysis on these MD simulations and ensure the outcomes are as expected.
 
     Returns
     -------
     Test the following:
 
-    * Two experiments added to a project successfully
+        * Two experiments added to a project successfully
+        * Correct coordination numbers computed
+        * Correct POMF values computed
+        * Dynamics run successfully.
 
     Notes
     -----
-    TODO: Add correct tests when all post-RDF calculators are fixed.
+    See the link below for similar data for CNs for molten salts.
+    https://link.springer.com/article/10.1007/s10800-018-1197-z
     """
-    pass
+    NaCl_experiment = mdsuite_project.experiments.NaCl
+    KCl_experiment = mdsuite_project.experiments.KCl
+
+    RDF_Data = mdsuite_project.run.RadialDistributionFunction(
+        number_of_configurations=500, cutoff=15.0, plot=False
+    )
+    NaCl_CN_data = NaCl_experiment.run.CoordinationNumbers(
+        rdf_data=RDF_Data["NaCl"],
+        savgol_window_length=111,
+        savgol_order=9,
+        number_of_shells=3,
+        plot=False,
+    )
+    KCl_CN_data = KCl_experiment.run.CoordinationNumbers(
+        rdf_data=RDF_Data["KCl"],
+        savgol_window_length=111,
+        savgol_order=7,
+        number_of_shells=2,
+        plot=False,
+    )
+    KCl_POMF_data = KCl_experiment.run.PotentialOfMeanForce(
+        rdf_data=RDF_Data["KCl"],
+        savgol_window_length=111,
+        savgol_order=7,
+        number_of_shells=2,
+        plot=False,
+    )
+    # Run assertions on selected observables
+    NaCl_CN_data["Na_Cl"]["CN_1"] == pytest.approx(5.213, 0.0001)
+    NaCl_CN_data["Na_Cl"]["CN_2"] == pytest.approx(35.090, 0.0001)
+    NaCl_CN_data["Na_Na"]["CN_1"] == pytest.approx(14.775, 0.0001)
+
+    KCl_CN_data["Cl_K"]["CN_1"] == pytest.approx(5.507, 0.0001)
+
+    KCl_POMF_data["Cl_K"]["POMF_1"] == pytest.approx(-1.372e-11, 1e-14)
+
+    mdsuite_project.run.GreenKuboDiffusionCoefficients()
+    mdsuite_project.run.EinsteinDiffusionCoefficients()
```

Comment:
In principle I can add data but as was suggested in the previous tests I can only provide some references of papers with approximate values. Testing the complete diffusion value and msd is done in the proper LJ verified test. In case of GK we have not a proper test yet.